File: src/lib/server/validation/responses/sleep.ts
import 'zod-openapi';
import { z } from 'zod';

const sleepEntrySchema = z
	.object({
		id: z.string().uuid(),
		userId: z.string().uuid(),
		entryDate: z.string(),
		durationMinutes: z.number().int(),
		quality: z.number().int(),
		bedtime: z.string().nullable(),
		wakeTime: z.string().nullable(),
		wakeUps: z.number().int().nullable(),
		sleepLatencyMinutes: z.number().int().nullable(),
		deepSleepMinutes: z.number().int().nullable(),
		lightSleepMinutes: z.number().int().nullable(),
		remSleepMinutes: z.number().int().nullable(),
		source: z.string().nullable(),
		notes: z.string().nullable(),
		loggedAt: z.string().optional(),
		createdAt: z.string().optional(),
		updatedAt: z.string().optional()
	})
	.meta({ id: 'SleepEntry' });

export const sleepEntriesResponseSchema = z
	.object({
		entries: z.array(sleepEntrySchema)
	})
	.meta({ id: 'SleepEntriesResponse' });

export const sleepEntryResponseSchema = z
	.object({
		entry: sleepEntrySchema
	})
	.meta({ id: 'SleepEntryResponse' });

File: mobile/iosApp/Bissbilanz/Models/Sleep.swift
import Foundation

struct SleepEntry: Codable, Identifiable {
    let id: String
    let userId: String
    let entryDate: String
    let durationMinutes: Int
    let quality: Int
    let bedtime: String?
    let wakeTime: String?
    let wakeUps: Int?
    let sleepLatencyMinutes: Int?
    let deepSleepMinutes: Int?
    let lightSleepMinutes: Int?
    let remSleepMinutes: Int?
    let source: String?
    let notes: String?
    let loggedAt: String?
    let createdAt: String?
    let updatedAt: String?
}

struct SleepCreate: Codable {
    let durationMinutes: Int
    let quality: Int
    let entryDate: String
    var bedtime: String?
    var wakeTime: String?
    var wakeUps: Int?
    var notes: String?
}

struct SleepUpdate: Codable {
    var durationMinutes: Int?
    var quality: Int?
    var entryDate: String?
    var bedtime: String?
    var wakeTime: String?
    var wakeUps: Int?
    var notes: String?
}

struct SleepEntriesResponse: Codable {
    let entries: [SleepEntry]
}

struct SleepEntryResponse: Codable {
    let entry: SleepEntry
}
